slipcover/bytecode.py

```diff
@@ -629,44 +665,56 @@ def find_load_globals():
                 delta += change
 
 
-    def finish(self):
-        """Finishes editing bytecode, returning a new code object."""
+    def _finish(self):
+        if not self.finished:
+            self.finished = True
 
-        assert not self.finished
-        self.finished = True
+            if self.branches is not None:
+                # A branch's new target may now require more EXTENDED_ARG opcodes to be expressed.
+                # Inserting space for those may in turn trigger needing more space for others...
+                # FIXME missing test for length adjustment triggering other length adjustments
```

Comment:
Correctness issue?

Reply:
I am not sure what you are asking... this code replicates what I've seen Python code do, but I didn't put enough time into creating tests to successfully test out this scenario. That is what the FIXME is saying... that this isn't tested out.